### DISCUSSION TITLE
Z probe cycle parameters?

### DISCUSSION THREAD

**Discussion by stevempotter** [grblHAL/core#569] [Q&A] [answered]
I am setting the Z-zero with a touchplate. Sometimes, if I have thick stock or a long bit, there is not enough Z travel for the bit to be retracted after probing and it throws an Alarm 2.  Where or how can I set it so it only lifts a couple of mm instead of 10?

**Comment by terjeio** [comment, marked as answer]
Is this part of a tool change sequence (M6)? Using ioSender?
If so the retract distance before the slow probe is hardcoded to 2mm, main retract is back to Z home then the original tooltip position unless $346 _Restore position_ after M6 is set to 1/true. Or if an old controller build, pre mid 2021, then 4mm - depending on the tool change mode.
The 10mm retract is not by the controller itself, perhaps a macro, your gcode or something else?

**Reply by stevempotter** [reply]
Thanks, that is very helpful. Yes, it turns out it was in my own pendant code! (in the Receiver code, Buttons.h)
Problem solved.
